lib/utils.js

```diff
@@ -217,35 +216,29 @@ utils.AsyncWrapper = (function() {
      * Schedules a function call in the most immenent microtask.
      * This cannot be canceled.
      */
-    AsyncWrapper.prototype.runAsap = utils.MutationObserver ? function() {
+    AsyncWrapper.prototype.runAsap = function() {
+        if (this.asapScheduled) { return; }
+        this.asapScheduled = true;
+
         cAF(this.rAFid);
         clearTimeout(this.timerId);
-        /**
-         * Using MutationObservers to access microtask queue is a standard technique,
-         * used in ASAP library
-         * {@link https://github.com/kriskowal/asap/blob/master/browser-raw.js#L140}
-         */
-        if (!this.mo) {
-            this.mo = new utils.MutationObserver(this.wrappedCallback);
-            this.node = document.createTextNode(1);
-            this.mo.observe(this.node, { characterData: true });
+
+        if (utils.MutationObserver) {
+            /**
+             * Using MutationObservers to access microtask queue is a standard technique,
```

Comment:
Theoretical question: why not using `Promise` instead?

Reply:
MutationObserver callbacks fire earlier than Promise in some browsers, although it is not compliant to the spec

Reply:
Seems to be resolved in Safari btw: https://bugs.webkit.org/show_bug.cgi?id=147933
Not sure about Edge.

Reply:
It appears that it's now resolved in Edge as well:
https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/4091192/